.github/workflows/ci.yml

```diff
@@ -552,3 +552,35 @@ jobs:
           echo "Testing against ${RELAY_TEST_IMAGE}"
           make test-relay-integration
 
+  self-hosted-end-to-end:
+    runs-on: ubuntu-latest
+    timeout-minutes: 25
+    # temporary, remove once we are confident the action is working
+    continue-on-error: true
+    needs: build
+
+    # Skip redundant checks for library releases
+    if: "!startsWith(github.ref, 'refs/heads/release-library/')"
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+
+      - name: Download Docker Image
+        uses: actions/download-artifact@v3
+        with:
+          name: relay-docker-image
+
+      - name: Import Docker Image
+        run: docker load -i relay-docker-image.tgz
```

Comment:
Oh and that doesn't exist anymore

Reply:
I think the entire download and import is not necessary, do you even need the checkout of the repo?

Reply:
I don't think I'll need the checkout of the repo, will remove and verify